lib/parsers/package-lock-parser.ts

```diff
@@ -261,11 +294,18 @@ export class PackageLockParser implements LockfileParser {
         const subDep = depTrees[subDepPath];
         dep.dependencies[subDep.name] = subDep;
       }
-      delete dep.requires;
-      depTrees[depKey] = {...dep as PkgTree};
+      const pkgTree: PkgTree = {
+        cyclic: dep.cyclic,
+        depType: dep.depType,
+        dependencies: dep.dependencies,
+        hasDevDependencies: dep.hasDevDependencies,
+        name: dep.name,
+        version: dep.version,
+      };
+      depTrees[depKey] = pkgTree;
       // Since this code doesn't handle any I/O or network, we need to force
       // event loop to tick while being used in server for request processing
-      await setImmediatePromise();
+      await this.setImmediatePromise();
```

Comment:
it's a minor thing, but would personally prefer the `setImmediatePromise` to be a static or even global function as it doesn't use any internal state of the the class instance.